src/js/gi/selectors/calculator.js

```diff
@@ -9,15 +9,764 @@ const getEligibilityDetails = (state) => {
+  } else if (inputs.calendar === 'nontraditional' && numberOfTerms < 3) {
+    bookStipendTerm3 = 0;
+  } else if (oldGiBill) {
+    bookStipendTerm3 = 0;
+  } else if (your.giBillChapter === 31) {
+    bookStipendTerm3 = inputs.books / numberOfTerms;
+  } else {
+    bookStipendTerm3 = ropBook *
+      constant.BSCAP / numberOfTerms * tier;
+  }
+
+  // Calculate Book Stipend for Year - getBookStipendYear
+  if (isOJT && your.giBillChapter === 33) {
+    bookStipendTotal = constant.BSOJTMONTH;
+  } else {
+    bookStipendTotal = bookStipendTerm1 + bookStipendTerm2 + bookStipendTerm3;
+  }
+
+  // Calculate Total Payments to You - getTotalPaidToYou
+  const totalToYou = housingAllowTotal + bookStipendTotal;
+
+  // Calculate Total Benefits for Term 1 - getTotalTerm1
+  if (isOJT) {
+    totalTerm1 = 0;
+  } else {
+    totalTerm1 = tuitionFeesTerm1 + yrBenTerm1 + housingAllowTerm1 + bookStipendTerm1;
+  }
+
+  // getTotalTerm2
+  if (inputs.calendar === 'nontraditional' && numberOfTerms === 1) {
+    bookStipendTerm2 = 0;
+  } else if (isOJT) {
+    totalTerm2 = 0;
+  } else {
+    totalTerm2 = tuitionFeesTerm2 +
+      yrBenTerm2 + housingAllowTerm2 +
+      bookStipendTerm2;
+  }
+
+  // getTotalTerm3
+  if (inputs.calendar === 'semesters') {
+    totalTerm3 = 0;
+  } else if (inputs.calendar === 'nontraditional' && numberOfTerms < 3) {
+    totalTerm3 = 0;
+  } else if (isOJT) {
+    totalTerm3 = 0;
+  } else {
+    totalTerm3 = tuitionFeesTerm3 +
+      yrBenTerm3 + housingAllowTerm3 +
+      bookStipendTerm3;
+  }
+
+  // Calculate Text for Total Benefits Row - getTotalText
+  if (your.giBillChapter === 33) {
+    giBillTotalText = 'Total Post-9/11 GI Bill Benefits';
+  } else if (your.giBillChapter === 30) {
+    giBillTotalText = 'Total Montgomery GI Bill Benefits';
+  } else if (your.giBillChapter === 1606) {
+    giBillTotalText = 'Total Select Reserve GI Bill Benefits';
+  } else if (your.giBillChapter === 1607) {
+    giBillTotalText = 'Total REAP GI Bill Benefits';
+  } else if (your.giBillChapter === 35) {
+    giBillTotalText = 'Total DEA GI Bill Benefits';
+  } else if (your.giBillChapter === 31) {
+    giBillTotalText = 'Total Voc Rehab Benefits';
+  }
+
+  // Calculate Total Benefits for Year - getTotalYear
+  if (isOJT) {
+    totalYear = 0;
+  } else {
+    totalYear = tuitionFeesTotal + yrBenTotal + housingAllowTotal + bookStipendTotal;
+  }
+
+  // Calculate Monthly Rate for Display - getMonthlyRateDisplay
+  if (isOJT) {
+    monthlyRateDisplay = housingAllowTerm1;
+  } else {
+    monthlyRateDisplay = housingAllowTerm1 / termLength;
+  }
+
+  return {
+    serviceDischarge,
+    vre911Eligible,
+    tier,
+    onlyVRE,
+    oldGiBill,
+    onlyTuitionFees,
+    monthlyRate,
+    numberOfTerms,
+    tuitionNetPrice,
+    housingStipdendTerm1,
+    housingStipdendTerm2,
+    housingStipdendTerm3,
+    housingStipdendTotal,
+    tuitionAllowTerm1,
+    tuitionAllowTerm2,
+    tuitionAllowTerm3,
+    tuitionAllowTotal,
```

Comment:
They are not constants. They are computed based on calculator form inputs and data from the backend (school attrs and constants). Basically what you are suggesting is akin to making the calculator a server-side activity requiring a round trip every time the user changes a form input.